agent/reconciler/host_reconciler.go

```diff
@@ -268,7 +252,7 @@ func (r *HostReconciler) hostCleanUp(ctx context.Context, byoHost *infrastructur
 		}
 		if r.SkipK8sInstallation {
 			logger.Info("Skipping uninstallation of k8s components")
-		} else if r.UseInstallerController {
+		} else {
 			if byoHost.Spec.UninstallationScript == nil {
```

Comment:
```suggestion
		} else if byoHost.Spec.UninstallationScript == nil {
```

Reply:
The condition `if byoHost.Spec.UninstallationScript == nil` ends at line 256 while the outer else ends at 271 so this will not work

Reply:
line#271 else is a part of its line#245 if condition. Isn’t it?

Reply:
Yes I meant 270 if we do as suggested then line 257 to 269 will become deadcode

Reply:
ooh ok got it. Then there is no need for else condition here. WDYT?

Reply:
Ok, my bad. I got confused by multiple if conditions.